integration_tests/tests/integration/simulation/reproduction.rs

```diff
@@ -0,0 +1,219 @@
+            min_aggregation_job_size: 1,
+            max_aggregation_job_size: 10,
+        },
+        ops: Vec::from([
+            Op::Upload {
+                report_time: START_TIME,
+            },
+            Op::AggregationJobCreator,
+            Op::AggregationJobDriver,
+            Op::LeaderGarbageCollector,
+            Op::Upload {
+                report_time: START_TIME,
+            },
+            Op::AggregationJobCreator,
+            Op::AggregationJobDriver,
+            Op::LeaderGarbageCollector,
+            Op::Upload {
+                report_time: START_TIME,
+            },
+            Op::AggregationJobCreator,
+            Op::AggregationJobDriver,
+            Op::LeaderGarbageCollector,
+            Op::CollectorStart {
+                collection_job_id,
+                query: Query::FixedSize(janus_messages::Query::new_fixed_size(
+                    FixedSizeQuery::CurrentBatch,
+                )),
+            },
+            Op::CollectionJobDriver,
+            Op::CollectorPoll { collection_job_id },
+            Op::Upload {
+                report_time: START_TIME,
+            },
+            Op::Upload {
+                report_time: START_TIME,
+            },
+            Op::Upload {
+                report_time: START_TIME,
+            },
+            Op::Upload {
+                report_time: START_TIME,
+            },
+            Op::AggregationJobCreator,
+            Op::AggregationJobDriver,
+            Op::CollectorStart {
+                collection_job_id,
+                query: Query::FixedSize(janus_messages::Query::new_fixed_size(
+                    FixedSizeQuery::CurrentBatch,
+                )),
+            },
+            Op::CollectionJobDriver,
+            Op::CollectorPoll { collection_job_id },
+        ]),
+    };
+    assert!(!Simulation::run(input).is_failure());
+}
+
+#[test]
+#[ignore = "failing test"]
+/// Reproduction of https://github.com/divviup/janus/issues/3323.
+fn repro_slow_uploads_with_max_batch_size() {
+    install_test_trace_subscriber();
+
+    let collection_job_id = random();
+    let input = Input {
+        is_fixed_size: true,
+        config: Config {
+            time_precision: Duration::from_seconds(3600),
+            min_batch_size: 4,
+            max_batch_size: Some(6),
+            report_expiry_age: Some(Duration::from_seconds(7200)),
+            min_aggregation_job_size: 1,
+            max_aggregation_job_size: 10,
+        },
+        ops: Vec::from([
+            Op::Upload {
+                report_time: START_TIME,
+            },
+            Op::AggregationJobCreator,
+            Op::AggregationJobDriver,
+            Op::AdvanceTime {
+                amount: Duration::from_seconds(3600),
+            },
+            Op::LeaderGarbageCollector,
+            Op::Upload {
+                report_time: Time::from_seconds_since_epoch(1_700_003_600),
+            },
```

Comment:
This is quite exciting, it kills tons of boilerplate when trying to reproduce production issues in tests.